keps/sig-node/2400-node-swap/README.md

```diff
@@ -300,6 +354,12 @@ and/or workloads in a number of different scenarios.
 Since swap provisioning is out of scope of this proposal, this enhancement
 poses low risk to Kubernetes clusters that will not enable swap.
 
+#### Security risk
+
+Enabling swap on a system without encryption poses a security risk, as critical information, such as Kubernetes secrets, may be swapped out to the disk. If an unauthorized individual gains access to the disk, they could potentially obtain these secrets. To mitigate this risk, it is recommended to use encrypted swap. However, handling encrypted swap is not within the scope of kubelet; rather, it is a general OS configuration concern and should be addressed at that level. Nevertheless, it is essential to provide documentation that warns users of this potential issue, ensuring they are aware of the potential security implications and can take appropriate steps to safeguard their system.
+
+Additionally end user may decide to disable swap completely for a Pod in beta 1 by making it guaranteed. This way, there will be no swap enabled for the corresponding containers and there will be no information exposure risks.
```

Comment:
How can we guarantee that system reserved can be set to zero so no system processes will ever use swap? Can we add this as an explicit goal?

Reply:
I'm not sure how we can guarantee that.
With @pacoxu's [PR](https://github.com/kubernetes/kubernetes/pull/105271) we can reserve some swap space to the system, but reserving 0 swap to the system does not mean that the system cannot use swap, it's just that there won't be a swap space that's dedicated to it.

I don't see a practical way of achieving this with cgroups, unless we can mess with the node's non-kubernetes cgroups hierarchies, and I assume that we can't do that.

@harche @pacoxu WDYT?

Reply:
@SergeyKanzhelev we can set `memory.swap.max` to `0` on system reserved cgroup (in our case usually that happens to be `/sys/fs/cgroup/system.slice`) and if we set `memory.swap.max` to `max` on burstable pods slice (which in our case usually happens to be `/sys/fs/cgroup/kubepods.slice/kubepods-burstable.slice`) we can 
make sure that no system processes will ever use swap.

As long as system reserved cgroup is setup in a way such that it doesn't end up becoming the parent of the  cgroups where burstable pods are running this arrangement will work fine. Because if someone configures their cgroup in a way where system cgroup is a parent of burstable pods cgroup then this will silently fail because a child cgroup cannot have `memory.swap.max` set to `max` while parent is set to `0`.